feat(Images): Change whole logic of receiving and storing images for projects

MartinMiklovic opened the pull request.

MartinMiklovic committed:
fix(image carousel) Fix image carousel in home page

MartinMiklovic committed:
fix(gallery) Fix spinners color in gallery

MartinMiklovic committed:
fix(gallery) Fix spinners color in gallery

MartinMiklovic committed:
feat(Images): Change whole logic of receiving and storing images for projects

MartinMiklovic committed:
Merge remote-tracking branch 'origin/HFA-130' into HFA-132

# Conflicts:
#	src/app/public-view/components/home/components/home-image-carousel/home-image-carousel.component.ts
#	src/environments/environment.prod.ts

RandomUserIK commented on src/app/admin-view/components/create-project/services/create-project-form-initializer.service.ts:
```diff
@@ -20,10 +22,10 @@ export class CreateProjectFormInitializerService {
+        this.initializePhotoGallery(formControl, projectData.galleryImages);
         break;
       case DataFieldType.DYNAMIC_TEXT_SECTION:
         // TODO: implement as NgxEditor
```
co s tymto TODO?

RandomUserIK commented on src/app/models/abstract-project-gallery.ts:
```diff
@@ -72,6 +71,7 @@ export abstract class AbstractProjectGalleryDirective extends AbstractGallery im
   }
 
   private processData(): (data: any) => void {
+    console.log('test');
```
vymazat

MartinMiklovic replied:
oh shit

RandomUserIK commented on src/app/components/image-carousel/image-carousel.component.ts:
```diff
@@ -26,11 +27,7 @@ export class ImageCarouselComponent implements OnInit {
 
   private prepareSafeUrlArrayForCarousel(): void {
```
kedze sa tu nastavuje iba property value tak by som zmenil aj nazov funkcie

MartinMiklovic replied:
V dalsom pr (HFA-135) ktory nadvazuje na tuto branchu je funkcia zmazana.

RandomUserIK commented on src/app/components/data-fields/services/file-upload-validation.service.ts:
```diff
@@ -3,19 +3,18 @@ import { Injectable } from '@angular/core';
 export class FileUploadValidationService {
 
-  private allowedFileTypes = ['image/png', 'image/jpeg'];
+  private allowedFileTypes = ['image/png', 'image/jpeg', 'image/webp'];
```
tu by sme este mali povolit aj jpg a bmp

RandomUserIK commented on src/app/components/data-fields/components/image-field/image-field.component.ts:
```diff
@@ -13,55 +14,52 @@ export class ImageFieldComponent implements OnInit {
+    reader.readAsDataURL(imageFile);
     reader.onload = () => {
-      this.dataField.imgSrc = this.sanitizer.bypassSecurityTrustUrl(reader.result as string) // NOSONAR
+      this.imageUrl = reader.result as string; // NOSONAR
```
tu treba teraz NOSONAR?

RandomUserIK commented on src/app/public-view/components/content/components/content-individual-project-photos/content-individual-project-photos.component.ts:
```diff
@@ -22,7 +22,8 @@ export class ContentIndividualProjectPhotosComponent implements OnInit {
     individualProjects.forEach((value: Project) => {
-      photoFileObservables.push(this.fileService.getFileFromPathAsSafeUrl(value.titleImage));
+      // todo
+      // photoFileObservables.push(this.fileService.getFileFromPathAsSafeUrl(value.titleImagePath)); todo fix
```
toto by sme tiez v ramci tohto PR mali poriesit

MartinMiklovic replied:
V dalsom pr (HFA-135) je to poriesene.

RandomUserIK commented on src/app/public-view/components/home/components/home-image-carousel/home-image-carousel.component.ts:
```diff
@@ -9,18 +9,17 @@ import {CarouselProject} from '../../models/CarouselProject';
-          this.carouselItems = []
           response.projects.forEach(value => {
-            this.carouselItems.push({id: value.id, image: value.titleImage, title: value.title})
+            // this.carouselItems.push({id: value.id, image: value.titleImagePath, title: value.title}) todo fix
```
prijebany carousel :hankey:

MartinMiklovic replied:
V dalsom pr (HFA-135) je to poriesene.

Commit:
refactor(blog): refactor blog article logic

trofoSK commented on src/app/admin-view/components/create-project/services/create-project-form-initializer.service.ts:
```diff
@@ -20,10 +22,10 @@ export class CreateProjectFormInitializerService {
         break;
       case DataFieldType.DYNAMIC_PHOTO_GALLERY:
-        this.initializePhotoGallery(formControl, projectData[dataField.formControlName]);
+        this.initializePhotoGallery(formControl, projectData.galleryImages);
```
tu vytahujeme data inac nez v nasledujucich casoch, zjednotme to

trofoSK commented on src/app/components/search-header/services/search-header.service.ts:
```diff
@@ -1,12 +1,12 @@
 export class SearchHeaderService {
 
-  private _searchHeaderState: BehaviorSubject<string> = new BehaviorSubject<string>('');
+  private _searchHeaderState: Subject<string> = new Subject<string>();
```
Preco tato zmena ?

MartinMiklovic replied:
Nechceme defaultne posielat prazdny string. Pokial by to bolo ako BehaviorSubject pri inicializovani galerie by zbytocne dva krat po sebe volal request na nacitanie vsetkych obrazkov na prvej strane co je zbytocne.

Commit:
refactor(blog):

- fix blog article component initialization
- refactor blog article create/update logic

MartinMiklovic committed:
Merge remote-tracking branch 'origin/HFA-132' into HFA-132

trofoSK commented on src/app/public-view/components/home/components/home-image-carousel/home-image-carousel.component.ts:
```diff
@@ -1,29 +1,11 @@
-      )
-  }
+export class HomeImageCarouselComponent {
+  images = ['assets/slide1.png', 'assets/slide2.png'];
```
toto nechame zatial takto ?

MartinMiklovic replied:
V dalsom pr (HFA-135) je to poriesene.

MartinMiklovic committed:
fix(Light box): Apply comments form pr

MartinMiklovic committed:
fix(Light box): Apply comments form pr
